vowpalwabbit/v_array.h

```diff
@@ -43,7 +43,25 @@ struct v_array
     _begin = nullptr;
     _end = nullptr;
     end_array = nullptr;
-    erase_count = 0;
+    _erase_count = 0;
+  }
+
+  void reserve_nocheck(size_t length)
+  {
+    if (capacity() == length) { return; }
```

Comment:
`if(capacity() == length || length == 0)`, and remove the length check on line 55 (realloc for size 0 is undefined behavior)

Reply:
Should we add a static assert for sizeof(T) > 0? Since the realloc function is undefined for 0?